### ISSUE TITLE
Evaluation with Detic in simulation

### ISSUE THREAD

**Issue by Sukiku** [allenai/PoliFormer#5]
Thank you for sharing this great project! I have reproduced your results with the pretrained models of pure text-nav, pure box-nav and text+box-nav, but I found some confusing result when I evaluate with Detic in simulation.

In the paper, it is stated that ground truth bounding box is used in simulation, while Detic is used as a substitute in reality. The addition of ground truth bounding box improves the results in simulation (85% to over 92%). The addition of Detic also makes certain improvements to pure text-nav in reality (83.3% to 88.9%). In simulation, however, it seems that replacing the ground truth bounding box with that obtained by Detic leads to a significant performance drop compared with pure text.

When I change the argument --det_type from "gt" to "detic", the success rate decrease to about 57% for both pure box-nav and text+box-nav settings, which is lower than using pure text as input (85%). Using Detic in simulation is not reported in the paper. If my results are correct, why is there such a big gap between reality and simulation?

Thank you for your response.

**Comment by KuoHaoZeng** [comment]
Hi! Thank you for the interests in our work!

Yes, your simulation results are correct-we've observed similar results on our side. We have several observations:
1. Our box-nav (or text+box-nav) policy is trained with ground truth bounding boxes from the simulator. It is sensitive to bounding box signal. For example, when a box is detected "robustly" (i.e., visible with continuous meaningful coordinates over time), the agent beelines to get closer to the detected box. We found that when using Detic in simulation, the box detection is not robust enough; the object might not be detected in some poses. It is possibly because Detic is trained fully on real-world images and never finetuned in simulation. In such cases, we found that the policy becomes confused. Interestingly, the policy tries to adjust the agent's pose around the object to see if it can regain the box signal.
2. Again, possibly due to Detic never being finetuned in simulation, some target objects are never successfully detected in our simulation benchmark. In that case, the box-nav policy is not possible to issue the "Done" action.
3. To improve it, we actually tried replacing Detic by Owl as well as VLM (Qwen-VL). For Owl, we found the performance is sightly worse than Detic. For the VLM, we tried using GPT4o + Qwen-VL: we first prompt GPT4o to check if a target object (with attribute like "a red color apple") is in the view. If yes, we prompt Qwen-VL to predict the box. This version slightly improves the performance to ~6x%, but it still lags significantly behind the text-nav and ground truth box-nav policies.

For the real-world experiments, we found that Detic works really well most of the time. Few examples:

"Find an apple": Detic works quit robustly.
https://github.com/user-attachments/assets/4a13cd3a-94c2-4258-acb5-53f170c16fca

"Follow a RC car": Detic works very robustly.
https://github.com/user-attachments/assets/9cc578a3-3f13-47e5-a443-1d382e34ef6c

"Find a houseplant": Again, Detic works quit robustly, but since we always choose the largest box, the detection becomes the other houseplant in the end. In addition, in the final few frames, you can see that the detected box disappears, and the policy tries to adjust the agent's pose to regains the box signal. 
https://github.com/user-attachments/assets/335b63ca-3f79-42ff-a01d-3f513d10a596

Failure case "Find a bed", Detic misrecognizes a sofa with a pillow on it as a bed, so the policy fails.
https://github.com/user-attachments/assets/b9b38168-6f73-4987-b0d7-865fd9e8e585

To sum up, I believe the unsatisfactory results using Detic for box-nav (or text+box-nav) in simulation are due to the real-to-sim gap introduced by the object detector. It is reasonable since Detic was never trained on images from our simulation. It would be interesting to explore the following directions: (1) finetune Detic using simulation data, (2) train PoliFormer with noisy box signals. or (3) train PoliFormer with Detic (if the inference speed of Detic is acceptable).

**Comment by Sukiku** [comment]
Many thanks for your reply!
